extensions/email/src/main/java/org/mule/extension/email/internal/commands/PagingProviderEmailDelegate.java

```diff
@@ -152,17 +150,19 @@ private MultiPartPayload readContent(javax.mail.Message m) {
         endIndex = min(endIndex + pageSize, folder.getMessageCount());
 
         if (!emails.isEmpty()) {
-          if (deleteAfterRetrieve) {
-            expungeCommand.expunge(connection, folderName);
-          }
           return emails;
         }
       }
     } catch (MessagingException e) {
       throw new EmailException("Error while retrieving emails: ", e);
+    } finally {
+      if (deleteAfterRetrieve) {
```

Comment:
This logic is not equivalent to the prior one. What should happen on the finally is the closeFolder(). But the expunge thing should only happen if deleteAfterRetrieve AND the result came back empty. 

With this change you always expunge which is wrong.